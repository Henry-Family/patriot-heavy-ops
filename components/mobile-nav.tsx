import * as React from "react"
import Link from "next/link"
import Image from "next/image"

import { MainNavItem } from "types"
// Removed unused siteConfig import
import { cn } from "@/lib/utils"
import { useLockBody } from "@/hooks/use-lock-body"
import { Icons } from "@/components/icons"
import { buttonVariants } from "@/components/ui/button"

type MobileNavProps = {
  items: MainNavItem[]
  children?: React.ReactNode
  onClose: () => void
}

export function MobileNav({ items, children, onClose }: MobileNavProps) {
  useLockBody()

  return (
    <div
      className={cn(
        "fixed inset-0 z-50 flex justify-end"
      )}
    >
      {/* Backdrop */}
      <div 
        className="absolute inset-0 bg-black/50 backdrop-blur-sm"
        onClick={onClose}
        onKeyDown={(e) => e.key === 'Enter' && onClose()}
        role="button"
        tabIndex={0}
      />
      
      {/* Menu Panel - Slides in from right */}
      <div className="relative z-20 size-full max-w-sm bg-white shadow-2xl duration-300 animate-in slide-in-from-right">
        {/* Logo and Close Button */}
        <div className="flex items-center justify-between border-b border-gray-200 p-6">
          <Link href="/" className="flex items-center space-x-2" onClick={onClose}>
            <Image
              src="/images/logo-color-h.webp"
              alt="Patriot Heavy Ops"
              width={150}
              height={30}
              className="h-auto w-[150px] object-contain"
            />
          </Link>
          <button onClick={onClose} className="rounded-md p-2 transition-colors hover:bg-gray-100">
            <Icons.close className="size-5" />
          </button>
        </div>
        
        {/* Navigation Content - Full height with scroll */}
        <div className="flex h-[calc(100vh-80px)] flex-col overflow-y-auto">
          {/* Navigation Items */}
          <nav className="flex-1 p-6">
            <div className="grid grid-flow-row auto-rows-max space-y-2 text-sm">
              {items.slice(0, 3).map((item, index) => (
                <Link
                  key={index}
                  href={item.disabled ? "#" : item.href}
                  className={cn(
                    "flex w-full items-center rounded-md p-3 text-sm font-medium text-black transition-colors hover:bg-gray-100",
                    item.disabled && "cursor-not-allowed opacity-60"
                  )}
                  onClick={onClose}
                >
                  {item.title}
                </Link>
              ))}
              
              {/* Separator with equal spacing */}
              <div className="my-8">
                <div className="border-t border-gray-200"></div>
              </div>
              
              {/* About Us and Contact Us */}
              {items.slice(3, 5).map((item, index) => (
                <Link
                  key={index + 3}
                  href={item.disabled ? "#" : item.href}
                  className={cn(
                    "flex w-full items-center rounded-md p-3 text-sm font-medium text-black transition-colors hover:bg-gray-100",
                    item.disabled && "cursor-not-allowed opacity-60"
                  )}
                  onClick={onClose}
                >
                  {item.title}
                </Link>
              ))}
              
              {/* Login and Sign Up Buttons */}
              <div className="space-y-3">
                <Link
                  href="/login"
                  className={cn(
                    buttonVariants({ variant: "outline", size: "sm" }),
                    "w-full justify-center text-black hover:text-black"
                  )}
                  onClick={onClose}
                >
                  Login
                </Link>
                <Link
                  href="/register"
                  className={cn(
                    buttonVariants({ size: "sm" }),
                    "w-full justify-center"
                  )}
                  onClick={onClose}
                >
                  Sign Up
                </Link>
              </div>
            </div>
          </nav>
          
          {children}
        </div>
      </div>
    </div>
  )
}
