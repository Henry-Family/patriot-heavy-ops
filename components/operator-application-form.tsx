"use client"

import * as React from "react"
import { useRouter } from "next/navigation"
import { zodResolver } from "@hookform/resolvers/zod"
import { User } from "@prisma/client"
import { useForm } from "react-hook-form"
import * as z from "zod"

import { cn } from "@/lib/utils"
import { operatorApplicationSchema } from "@/lib/validations/user"
import { buttonVariants } from "@/components/ui/button"
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from "@/components/ui/card"
import { Input } from "@/components/ui/input"
import { Label } from "@/components/ui/label"
import { toast } from "@/components/ui/use-toast"
import { Icons } from "@/components/icons"

interface OperatorApplicationFormProps
  extends React.HTMLAttributes<HTMLFormElement> {
  user: Pick<User, "id" | "name">
}

type FormData = z.infer<typeof operatorApplicationSchema>

interface AddressSuggestion {
  display_name: string
  lat: string
  lon: string
  place_id: string
}

export function OperatorApplicationForm({
  user,
  className,
  ...props
}: OperatorApplicationFormProps) {
  const router = useRouter()
  const {
    handleSubmit,
    setValue,
    formState: { errors },
  } = useForm<FormData>({
    resolver: zodResolver(operatorApplicationSchema),
    defaultValues: {
      location: "",
    },
  })
  const [isSaving, setIsSaving] = React.useState<boolean>(false)
  const [inputValue, setInputValue] = React.useState("")
  const [suggestions, setSuggestions] = React.useState<AddressSuggestion[]>([])
  const [isLoading, setIsLoading] = React.useState(false)
  const debounceRef = React.useRef<NodeJS.Timeout>()

  // const selectedLocation = watch("location")

  // Debounced address search
  const searchAddresses = React.useCallback(async (query: string) => {
    if (query.length < 3) {
      setSuggestions([])
      return
    }

    setIsLoading(true)
    try {
      const url = `/api/geocoding?q=${encodeURIComponent(query)}`

      const response = await fetch(url)
      const data = await response.json()

      setSuggestions(data || [])
    } catch {
      setSuggestions([])
    } finally {
      setIsLoading(false)
    }
  }, [])

  // Handle input change with debouncing
  const handleInputChange = (value: string) => {
    setInputValue(value)

    if (debounceRef.current) {
      clearTimeout(debounceRef.current)
    }

    debounceRef.current = setTimeout(() => {
      searchAddresses(value)
    }, 300)
  }

  // Handle address selection
  const handleAddressSelect = (suggestion: AddressSuggestion) => {
    const formattedAddress = suggestion.display_name
    setInputValue(formattedAddress)
    setValue("location", formattedAddress)
    setSuggestions([]) // Clear suggestions to hide dropdown
  }

  async function onSubmit(data: FormData) {
    setIsSaving(true)

    const response = await fetch(`/api/users/${user.id}/operator-application`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        location: data.location,
      }),
    })

    setIsSaving(false)

    if (!response?.ok) {
      return toast({
        title: "Something went wrong.",
        description: "Your application was not submitted. Please try again.",
        variant: "destructive",
      })
    }

    toast({
      description: "Your operator application has been submitted for review.",
    })

    router.refresh()
    return
  }

  return (
    <form
      className={cn(className)}
      onSubmit={handleSubmit(onSubmit)}
      {...props}
    >
      <Card>
        <CardHeader>
          <CardTitle>Apply to Become an Operator</CardTitle>
          <CardDescription>
            Are you a heavy equipment operator? Enter your service area so we
            can show job requests near you.
          </CardDescription>
        </CardHeader>
        <CardContent>
          <div className="grid gap-1">
            <Label htmlFor="location">Service Area</Label>
            <div className="relative">
              <Input
                value={inputValue}
                onChange={(e) => handleInputChange(e.target.value)}
                onBlur={() => {
                  // Hide dropdown after a short delay to allow clicks on suggestions
                  setTimeout(() => setSuggestions([]), 150)
                }}
                placeholder="Enter city, state, or address..."
                className="w-[400px]"
              />
              {isLoading && (
                <Icons.spinner className="absolute right-3 top-3 size-4 animate-spin" />
              )}
              {suggestions.length > 0 && (
                <div className="absolute z-50 mt-1 max-h-60 w-[400px] overflow-auto rounded-md border border-gray-200 bg-white shadow-lg">
                  {suggestions.map((suggestion) => (
                    <div
                      key={suggestion.place_id}
                      role="button"
                      tabIndex={0}
                      onClick={() => handleAddressSelect(suggestion)}
                      onKeyDown={(e) => {
                        if (e.key === "Enter" || e.key === " ") {
                          e.preventDefault()
                          handleAddressSelect(suggestion)
                        }
                      }}
                      className="cursor-pointer border-b border-gray-100 px-3 py-2 last:border-b-0 hover:bg-gray-100"
                    >
                      <div className="text-sm font-medium">
                        {suggestion.display_name.split(",")[0]}
                      </div>
                      <div className="truncate text-xs text-gray-500">
                        {suggestion.display_name}
                      </div>
                    </div>
                  ))}
                </div>
              )}
            </div>
            {errors?.location && (
              <p className="px-1 text-xs text-red-600">
                {errors.location.message}
              </p>
            )}
          </div>
        </CardContent>
        <CardFooter>
          <button
            type="submit"
            className={cn(buttonVariants(), className)}
            disabled={isSaving}
          >
            {isSaving && <Icons.spinner className="mr-2 size-4 animate-spin" />}
            <span>Apply Now</span>
          </button>
        </CardFooter>
      </Card>
    </form>
  )
}
