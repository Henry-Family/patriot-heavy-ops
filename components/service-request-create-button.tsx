"use client"

import * as React from "react"
import { useRouter } from "next/navigation"

import { cn } from "@/lib/utils"
import { ButtonProps, buttonVariants } from "@/components/ui/button"
import { toast } from "@/components/ui/use-toast"
import { Icons } from "@/components/icons"

type ServiceRequestCreateButtonProps = ButtonProps

export function ServiceRequestCreateButton({
  className,
  variant,
  ...props
}: ServiceRequestCreateButtonProps) {
  const router = useRouter()
  const [isLoading, setIsLoading] = React.useState<boolean>(false)

  async function onClick() {
    setIsLoading(true)

    try {
      // Navigate to the create service request page
      await router.push("/dashboard/requests/new")
    } catch {
      // Navigation error
      toast({
        title: "Navigation failed",
        description: "Unable to navigate to the create request page. Please try again.",
        variant: "destructive",
      })
    } finally {
      // Keep loading state until component unmounts (navigation completes)
      // Don't set loading to false here as the component will unmount
    }
  }

  return (
    <button
      onClick={onClick}
      className={cn(
        buttonVariants({ variant }),
        {
          "cursor-not-allowed opacity-60": isLoading,
        },
        className
      )}
      disabled={isLoading}
      {...props}
    >
      {isLoading ? (
        <Icons.spinner className="mr-2 size-4 animate-spin" />
      ) : (
        <Icons.add className="mr-2 size-4" />
      )}
      New Service Request
    </button>
  )
}
