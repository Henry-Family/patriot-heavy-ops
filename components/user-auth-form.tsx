"use client"

import * as React from "react"
import { useSearchParams } from "next/navigation"
import { zodResolver } from "@hookform/resolvers/zod"
import { signIn } from "next-auth/react"
import { useForm } from "react-hook-form"
import type { FieldErrors } from "react-hook-form"
import * as z from "zod"

import { cn } from "@/lib/utils"
import { userLoginSchema, userRegisterSchema } from "@/lib/validations/auth"
import { buttonVariants } from "@/components/ui/button"
import { Input } from "@/components/ui/input"
import { Label } from "@/components/ui/label"
import { toast } from "@/components/ui/use-toast"
import { Icons } from "@/components/icons"

type UserAuthFormProps = React.HTMLAttributes<HTMLDivElement> & {
  mode?: "login" | "register"
}

type LoginFormData = z.infer<typeof userLoginSchema>
type RegisterFormData = z.infer<typeof userRegisterSchema>

export function UserAuthForm({
  className,
  mode = "login",
  ...props
}: UserAuthFormProps) {
  const isRegisterMode = mode === "register"
  const schema = isRegisterMode ? userRegisterSchema : userLoginSchema

  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<LoginFormData | RegisterFormData>({
    resolver: zodResolver(schema),
  })
  const [isLoading, setIsLoading] = React.useState<boolean>(false)
  const [isGitHubLoading, setIsGitHubLoading] = React.useState<boolean>(false)
  const searchParams = useSearchParams()

  async function onSubmit(data: LoginFormData | RegisterFormData) {
    setIsLoading(true)

    try {
      if (isRegisterMode) {
        // Handle registration
        const registerData = data as RegisterFormData
        const response = await fetch("/api/auth/register", {
          method: "POST",
          headers: { "Content-Type": "application/json" },
          body: JSON.stringify({
            name: registerData.name,
            email: registerData.email,
            password: registerData.password,
            confirmPassword: registerData.confirmPassword,
          }),
        })

        if (!response.ok) {
          const error = await response.json()
          throw new Error(error.message || "Registration failed")
        }

        // Automatically sign in the user after successful registration
        const signInResult = await signIn("credentials", {
          email: registerData.email.toLowerCase(),
          password: registerData.password,
          redirect: true,
          callbackUrl: searchParams?.get("from") || "/dashboard",
        })

        if (signInResult?.error) {
          // If auto-login fails, show success message and let user manually login
          toast({
            title: "Account created",
            description: "You can now sign in with your email and password.",
          })
        }
      } else {
        // Handle login
        const loginData = data as LoginFormData
        const signInResult = await signIn("credentials", {
          email: loginData.email.toLowerCase(),
          password: loginData.password,
          redirect: true,
          callbackUrl: searchParams?.get("from") || "/dashboard",
        })

        if (signInResult?.error) {
          throw new Error("Invalid email or password")
        }
      }
    } catch (error) {
      toast({
        title: "Something went wrong.",
        description:
          error instanceof Error ? error.message : "Please try again.",
        variant: "destructive",
      })
    }

    setIsLoading(false)
  }

  return (
    <div className={cn("grid gap-6", className)} {...props}>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="grid gap-2">
          {isRegisterMode && (
            <div className="grid gap-1">
              <Label className="sr-only" htmlFor="name">
                Name
              </Label>
              <Input
                id="name"
                placeholder="Full name"
                type="text"
                autoComplete="name"
                disabled={isLoading || isGitHubLoading}
                {...register("name")}
              />
              {isRegisterMode &&
                (errors as FieldErrors<RegisterFormData>)?.name && (
                  <p className="px-1 text-xs text-red-600">
                    {
                      (errors as FieldErrors<RegisterFormData>).name
                        ?.message as string
                    }
                  </p>
                )}
            </div>
          )}
          <div className="grid gap-1">
            <Label className="sr-only" htmlFor="email">
              Email
            </Label>
            <Input
              id="email"
              placeholder="name@example.com"
              type="email"
              autoCapitalize="none"
              autoComplete="email"
              autoCorrect="off"
              disabled={isLoading || isGitHubLoading}
              {...register("email")}
            />
            {errors?.email && (
              <p className="px-1 text-xs text-red-600">
                {errors.email.message}
              </p>
            )}
          </div>
          <div className="grid gap-1">
            <Label className="sr-only" htmlFor="password">
              Password
            </Label>
            <Input
              id="password"
              placeholder="Password"
              type="password"
              autoComplete={
                isRegisterMode ? "new-password" : "current-password"
              }
              disabled={isLoading || isGitHubLoading}
              {...register("password")}
            />
            {errors?.password && (
              <p className="px-1 text-xs text-red-600">
                {errors.password.message}
              </p>
            )}
            {isRegisterMode && (
              <div className="px-1 text-xs text-muted-foreground">
                <p>Password must be at least 12 characters with:</p>
                <ul className="ml-2 list-inside list-disc">
                  <li>One uppercase letter</li>
                  <li>One lowercase letter</li>
                  <li>One number</li>
                  <li>One special character</li>
                  <li>No common words or patterns</li>
                </ul>
              </div>
            )}
          </div>
          {isRegisterMode && (
            <div className="grid gap-1">
              <Label className="sr-only" htmlFor="confirmPassword">
                Confirm Password
              </Label>
              <Input
                id="confirmPassword"
                placeholder="Confirm password"
                type="password"
                autoComplete="new-password"
                disabled={isLoading || isGitHubLoading}
                {...register("confirmPassword")}
              />
              {isRegisterMode &&
                (errors as FieldErrors<RegisterFormData>)?.confirmPassword && (
                  <p className="px-1 text-xs text-red-600">
                    {
                      (errors as FieldErrors<RegisterFormData>).confirmPassword
                        ?.message as string
                    }
                  </p>
                )}
            </div>
          )}
          <button
            type="submit"
            className={cn(buttonVariants())}
            disabled={isLoading}
          >
            {isLoading && (
              <Icons.spinner className="mr-2 size-4 animate-spin" />
            )}
            {isRegisterMode ? "Create Account" : "Sign In"}
          </button>
        </div>
      </form>
      <div className="relative hidden">
        <div className="absolute inset-0 flex items-center">
          <span className="w-full border-t" />
        </div>
        <div className="relative flex justify-center text-xs uppercase">
          <span className="bg-background px-2 text-muted-foreground">
            Or continue with
          </span>
        </div>
      </div>
      <button
        type="button"
        className={cn(buttonVariants({ variant: "outline" }), "hidden")}
        onClick={() => {
          setIsGitHubLoading(true)
          signIn("github")
        }}
        disabled={isLoading || isGitHubLoading}
      >
        {isGitHubLoading ? (
          <Icons.spinner className="mr-2 size-4 animate-spin" />
        ) : (
          <Icons.gitHub className="mr-2 size-4" />
        )}{" "}
        Github
      </button>
    </div>
  )
}
