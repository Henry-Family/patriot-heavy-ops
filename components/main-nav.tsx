"use client"

import * as React from "react"
import Link from "next/link"
import { useSelectedLayoutSegment } from "next/navigation"

import { MainNavItem } from "types"
import { siteConfig } from "@/config/site"
import { cn } from "@/lib/utils"
import { Icons } from "@/components/icons"
import { MobileNav } from "@/components/mobile-nav"

interface MainNavProps {
  items?: MainNavItem[]
  children?: React.ReactNode
}

export function MainNav({ items, children }: MainNavProps) {
  const segment = useSelectedLayoutSegment()
  const [showMobileMenu, setShowMobileMenu] = React.useState<boolean>(false)

  return (
    <>
      {/* Hamburger Menu - Works on all screen sizes */}
      <button
        className="flex items-center space-x-2 text-white"
        onClick={() => setShowMobileMenu(!showMobileMenu)}
      >
        {showMobileMenu ? <Icons.close /> : <Icons.menu />}
      </button>
      
      {/* Mobile Menu Overlay - Works on all screen sizes */}
      {showMobileMenu && items && (
        <MobileNav items={items} onClose={() => setShowMobileMenu(false)}>
          {children}
        </MobileNav>
      )}
    </>
  )
}
