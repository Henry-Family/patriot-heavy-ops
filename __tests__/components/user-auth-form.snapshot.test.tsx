import React from "react"
import { expectComponentSnapshot } from "../helpers/snapshot-helpers"
import { UserAuthForm } from "@/components/user-auth-form"

describe("UserAuthForm Snapshots", () => {
  it("matches snapshot in login mode", () => {
    expectComponentSnapshot(<UserAuthForm mode="login" />)
  })

  it("matches snapshot in register mode", () => {
    expectComponentSnapshot(<UserAuthForm mode="register" />)
  })
})


