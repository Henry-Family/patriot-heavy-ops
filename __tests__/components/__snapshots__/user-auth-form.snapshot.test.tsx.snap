// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`UserAuthForm Snapshots matches snapshot in login mode 1`] = `
<div
  class="grid gap-6"
>
  <form>
    <div
      class="grid gap-2"
    >
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="email"
        >
          Email
        </label>
        <input
          autocapitalize="none"
          autocomplete="email"
          autocorrect="off"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="email"
          name="email"
          placeholder="name@example.com"
          type="email"
        />
      </div>
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="current-password"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Sign In
      </button>
    </div>
  </form>
  <div
    class="relative hidden"
  >
    <div
      class="absolute inset-0 flex items-center"
    >
      <span
        class="w-full border-t"
      />
    </div>
    <div
      class="relative flex justify-center text-xs uppercase"
    >
      <span
        class="bg-background px-2 text-muted-foreground"
      >
        Or continue with
      </span>
    </div>
  </div>
  <button
    class="items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background border border-input hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4 hidden"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="mr-2 size-4"
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
      />
    </svg>
     
    Github
  </button>
</div>
`;

exports[`UserAuthForm Snapshots matches snapshot in register mode 1`] = `
<div
  class="grid gap-6"
>
  <form>
    <div
      class="grid gap-2"
    >
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="name"
        >
          Name
        </label>
        <input
          autocomplete="name"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="name"
          name="name"
          placeholder="Full name"
          type="text"
        />
      </div>
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="email"
        >
          Email
        </label>
        <input
          autocapitalize="none"
          autocomplete="email"
          autocorrect="off"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="email"
          name="email"
          placeholder="name@example.com"
          type="email"
        />
      </div>
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="password"
        >
          Password
        </label>
        <input
          autocomplete="new-password"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
        <div
          class="px-1 text-xs text-muted-foreground"
        >
          <p>
            Password must be at least 12 characters with:
          </p>
          <ul
            class="ml-2 list-inside list-disc"
          >
            <li>
              One uppercase letter
            </li>
            <li>
              One lowercase letter
            </li>
            <li>
              One number
            </li>
            <li>
              One special character
            </li>
            <li>
              No common words or patterns
            </li>
          </ul>
        </div>
      </div>
      <div
        class="grid gap-1"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 sr-only"
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <input
          autocomplete="new-password"
          class="flex h-10 w-full rounded-md border border-input bg-transparent px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
          id="confirmPassword"
          name="confirmPassword"
          placeholder="Confirm password"
          type="password"
        />
      </div>
      <button
        class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background bg-primary text-primary-foreground hover:bg-primary/90 h-10 py-2 px-4"
        type="submit"
      >
        Create Account
      </button>
    </div>
  </form>
  <div
    class="relative hidden"
  >
    <div
      class="absolute inset-0 flex items-center"
    >
      <span
        class="w-full border-t"
      />
    </div>
    <div
      class="relative flex justify-center text-xs uppercase"
    >
      <span
        class="bg-background px-2 text-muted-foreground"
      >
        Or continue with
      </span>
    </div>
  </div>
  <button
    class="items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background border border-input hover:bg-accent hover:text-accent-foreground h-10 py-2 px-4 hidden"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="mr-2 size-4"
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
      />
    </svg>
     
    Github
  </button>
</div>
`;
